packages/components/rating/src/Rating.test.tsx

```diff
@@ -1,12 +1,127 @@
+import { fireEvent } from '@testing-library/dom'
 import { render, screen } from '@testing-library/react'
-import { describe, expect, it } from 'vitest'
+import userEvent from '@testing-library/user-event'
+import { useState } from 'react'
+import { describe, expect, it, vi } from 'vitest'
 
 import { Rating } from './Rating'
 
+const utils = {
+  getInput: () =>
+    screen.getByRole('slider', {
+      name: /rating/i,
+    }),
+  getStars: (container: HTMLElement) =>
+    Array.from(container.querySelectorAll('[data-part="star"]')),
+}
+
 describe('Rating', () => {
   it('should render', () => {
-    render(<Rating />)
+    const { container } = render(<Rating defaultValue={2} />)
+
+    expect(container.querySelector('[data-spark-component=rating]')).toBeInTheDocument()
+    expect(container.querySelectorAll('[data-part="star"]').length).toBe(5)
+  })
+
+  it('should be possible to interact', async () => {
+    const user = userEvent.setup()
+
+    const { container } = render(<Rating size="lg" defaultValue={1} />)
+
+    const stars = utils.getStars(container)
+    const input = utils.getInput()
+    const firstStar = stars.at(0)
+    const thirdStar = stars.at(2)
+
+    if (!thirdStar || !firstStar) throw new Error('Stars element not found')
```

Comment:
Why having this line in our UT? If L33 or L34 are falsy test would fail, and we also eventually could simply declare an expectation, wdyt? This syntax seems quite disturbing to me :thinking:

Reply:
> If L33 or L34 are falsy test would fail

Unfortunately, no. Even adding tests that assert that the star is indeed there will not make TS infer that it is present in the following code. But, we could always resort to using the non-null assertion operator 🤷